domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/casefield/CaseFieldEntityComplexFieldACLValidatorImpl.java

```diff
@@ -0,0 +1,109 @@
+    }
+
+    private void validateCRUDComplexParent(CaseFieldEntity caseField, CaseFieldEntityValidationContext caseFieldEntityValidationContext,
+                                           ValidationResult validationResult, ComplexFieldACLEntity complexFieldACLEntity) {
+        String userRole = complexFieldACLEntity.getUserRole() != null ? complexFieldACLEntity.getUserRole().getReference() : "";
+        caseField.getComplexFieldACLEntities()
+            .stream()
+            .anyMatch(entity -> {
+                boolean match = (entity.getUserRole() != null && entity.getUserRole().getReference().equalsIgnoreCase(userRole))
+                    && entity.getListElementCode().contains(complexFieldACLEntity.getListElementCode())
+                    && !entity.getListElementCode().equalsIgnoreCase(complexFieldACLEntity.getListElementCode())
+                    && complexFieldACLEntity.hasLessAccessThan(entity);
+                if (match) {
+                    validationResult.addError(new CaseFieldEntityComplexACLValidationError(
+                        String.format("List element code '%s' has more access than its parent '%s'", entity.getListElementCode(), complexFieldACLEntity.getListElementCode()),
+                        entity, new AuthorisationCaseFieldValidationContext(caseField, caseFieldEntityValidationContext)));
+                    LOG.info("List element code '{}' has more access than its parent '{}'", entity.getListElementCode(), complexFieldACLEntity.getListElementCode());
+                }
+                return match;
+            });
+        List<String> parentCodes = parseParentCodes(complexFieldACLEntity.getListElementCode());
+        final List<String> missingCodes = parentCodes
+            .stream()
+            .filter(parentCode -> missingInComplexACLs(caseField.getComplexFieldACLEntities(), userRole, parentCode))
+            .collect(Collectors.toList());
+        for (String code : missingCodes) {
+            validationResult.addError(new CaseFieldEntityComplexACLValidationError(
+                String.format("Parent list element code '%s' is missing for list element code '%s'", code, complexFieldACLEntity.getListElementCode()),
+                complexFieldACLEntity, new AuthorisationCaseFieldValidationContext(caseField, caseFieldEntityValidationContext)));
+            LOG.info("Parent list element code '{}' is missing for list element code '{}'", code, complexFieldACLEntity.getListElementCode());
+        }
+    }
+
+    private boolean missingInComplexACLs(List<ComplexFieldACLEntity> complexFieldACLEntities, String userRole, String parentCode) {
+        return complexFieldACLEntities.stream()
+            .noneMatch(entity -> (entity.getUserRole() != null && entity.getUserRole().getReference().equalsIgnoreCase(userRole))
+                && parentCode.equals(entity.getListElementCode())
+            );
+    }
+
+    List<String> parseParentCodes(String listElementCode) {
```

Comment:
I will be reusing this one soon in my next story. Will move it to utility class then.

Reply:
Done 👍